rllib/agents/dqn/dqn_tf_policy.py

```diff
@@ -378,51 +379,21 @@ def compute_q_values(policy: Policy,
     return value, logits, dist, state
 
 
-def _adjust_nstep(n_step: int, gamma: int, obs: TensorType,
-                  actions: TensorType, rewards: TensorType,
-                  new_obs: TensorType, dones: TensorType):
-    """Rewrites the given trajectory fragments to encode n-step rewards.
-
-    reward[i] = (
-        reward[i] * gamma**0 +
-        reward[i+1] * gamma**1 +
-        ... +
-        reward[i+n_step-1] * gamma**(n_step-1))
-
-    The ith new_obs is also adjusted to point to the (i+n_step-1)'th new obs.
-
-    At the end of the trajectory, n is truncated to fit in the traj length.
-    """
-
-    assert not any(dones[:-1]), "Unexpected done in middle of trajectory"
-
-    traj_length = len(rewards)
-    for i in range(traj_length):
-        for j in range(1, n_step):
-            if i + j < traj_length:
-                new_obs[i] = new_obs[i + j]
-                dones[i] = dones[i + j]
-                rewards[i] += gamma**j * rewards[i + j]
```

Comment:
interesting. I was looking at this, and I felt like this logic only writes data from i+j into i, where j>0. we then don't touch frame i anymore.
so somehow this felt ok to me even if obs and next_obs are coming from the same array.

```
>>> a = numpy.arange(10)
>>> a
array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
>>> for i in range(10):
...   for j in range(5):
...     if i + j < 10:
...       a[i] = a[i + j]
... 
>>> a
array([4, 5, 6, 7, 8, 9, 9, 9, 9, 9])
```

actually, any chance you can add a unit test for this failure case, where obs and next_obs are from the same underlying array?

Reply:
Great idea! Will add a test like that.

Reply:
done.